### PR TITLE
[BEAM-14317] Add an example JvmInitializer to org.apache.beam.examples

### PR DESCRIPTION
**Pull request by arunpandianp**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by arunpandianp** [comment]
R: @kennknowles

File: examples/java/src/main/resources/META-INF/services/org.apache.beam.sdk.harness.JvmInitializer [hunk 1]
```diff
@@ -0,0 +1 @@
+org.apache.beam.examples.common.ExampleJVMInitializer
```
Comment (kennknowles) [inline examples/java/src/main/resources/META-INF/services/org.apache.beam.sdk.harness.JvmInitializer:1]: You don't need to do this manually, I think.

File: examples/java/src/main/java/org/apache/beam/examples/common/ExampleJVMInitializer.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * limitations under the License.
+ */
+package org.apache.beam.examples.common;
+
+import org.apache.beam.sdk.harness.JvmInitializer;
+import org.apache.beam.sdk.options.PipelineOptions;
+
+// Currently, JvmInitializers are only supported in portable pipelines or when using Google Cloud
+// Dataflow.
+public class ExampleJVMInitializer implements JvmInitializer {
```
Comment (kennknowles) [inline examples/java/src/main/java/org/apache/beam/examples/common/ExampleJVMInitializer.java:25]: Can you use `@AutoService` to register this?

**Review by kennknowles — CHANGES_REQUESTED** [review]
Overall, I think this example would be most useful if it were integrated into a pipeline. Currently, it is just the same as automatic code that an IDE could create so it doesn't really show the user how they might use this.

**Comment by aaltay** [comment]
@arunpandianp - Could you please reply to the open comments?

File: examples/java/src/main/resources/META-INF/services/org.apache.beam.sdk.harness.JvmInitializer (hunk 1 above)
Comment (arunpandianp) [reply examples/java/src/main/resources/META-INF/services/org.apache.beam.sdk.harness.JvmInitializer:1]: Yes, if we use @AutoService this file will be generated automatically.

**Comment by arunpandianp** [comment]
@kennknowles
> I think this example would be most useful if it were integrated into a pipeline

The ExampleJVMInitializer is integrated and gets executed with the pipelines like org.apache.beam.examples.WordCount, org.apache.beam.examples.WindowedWordCount etc. The ServiceLoader uses the mapping from`resources/META-INF/services/org.apache.beam.sdk.harness.JvmInitializer` and loads ExampleJVMInitializer.

Are you suggesting to create a new gradle project with a sample pipeline, AutoService and a JVMInitializer? Where should it live?

**Pull request closed without merging**